### PR TITLE
Add PriceFilter and FilterApplication

### PR DESCRIPTION
**Pull request by hlb8122**
**Motivation**
Clients should be able set filters to prevent spam reaching their inbox.

**Alternatives**
Filters are hosted in the keyserver metadata.

### CODE REVIEW

File: filters.proto [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+message PriceFilter {
+    // Whether the price filter should be publicly viewable.
+    bool public = 1;
+    // Minimum price required for acceptance into inbox.
+    uint64 acceptance_price = 2;
+    // Minimum price required to trigger an inbox notification.
+    uint64 notification_price = 3;
+}
+
+// Collection of filters.
```
Comment (hlb8122) [inline filters.proto]: Allows for future filters to be added in a backwards compatible way. For example:
```protobuf
message Filters {
    // Price filter
    PriceFilter price_filter = 1;
    // Other filter
    SizeFilter size_filter = 2;
}
```
```

File: filters.proto [hunk 2]
```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+package filters;
+
+// Price filter to be applied to an inbox.
+message PriceFilter {
+    // Whether the price filter should be publicly viewable.
+    bool public = 1;
```
Comment (hlb8122) [inline filters.proto]: One might want to hide their filter settings.

File: filters.proto [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    // Minimum price required for acceptance into inbox.
+    uint64 acceptance_price = 2;
+    // Minimum price required to trigger an inbox notification.
+    uint64 notification_price = 3;
```
Comment (hlb8122) [inline filters.proto]: Only a certain price may trigger a SSE/WS event to clients. Distinct from acceptance to the servers inbox.

File: filters.proto (hunk 1 above)
Comment (hlb8122) [reply filters.proto]: We should also probably put some sort of nonce/timestamp here so re-use of signatures can't occur.

File: filters.proto (hunk 2 above)
Comment (hlb8122) [reply filters.proto]: Perhaps it'd be better to make this a submessage
```protobuf
message PriceFilter {
   Viewability viewability = 1;
   ...
}

message Viewability {
   ...
}
```
so we can extend/refine this later.

File: filters.proto [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+// Collection of filters.
+message Filters {
+    // Price filter
+    PriceFilter price_filter = 1;
+}
+
+// FilterApplication represents a request to apply an inbox filter. 
+message FilterApplication {
```
Comment (schancel) [inline filters.proto]: This payload is very similar to the payload here: https://github.com/cashweb/keyserver/blob/master/proto/addressmetadata.proto#L31

Is this something we can factor out as a common attribute to all these systems?

File: filters.proto (hunk 1 above)
Comment (schancel) [reply filters.proto]: I think more generally, there are some inbox settings, rather than filters.

File: filters.proto (hunk 3 above)
Comment (schancel) [reply filters.proto]: Is it important that these are outside the wallet?

File: filters.proto (hunk 2 above)
Comment (schancel) [reply filters.proto]: re these advertised out at all? I'm not sure we care? 🤔

**Pull request closed without merging**